Description: Following the same variable and struct usage within sys header
 file ucontext.h to cover powerpc 64 bit archs.
Author: Fernando Seiti Furusato <ferseiti@br.ibm.com>
--- sigsegv.c
+++ sigsegv.c
@@ -94,8 +94,11 @@ static void signal_segv(int signum, sigi
 #if !defined(__alpha__) && !defined(__ia64__) && !defined(__FreeBSD_kernel__) && !defined(__arm__) && !defined(__hppa__) && !defined(__sh__)
     for(i = 0; i < NGREG; i++)
         a2j_error("reg[%02d]       = 0x" REGFORMAT, i,
-#if defined(__powerpc__)
+#if defined(__powerpc__) && !defined(__powerpc64__)
                 ucontext->uc_mcontext.uc_regs[i]
+/* just following the variable type and member logic as the ones used here to cover ppc64*/
+#elif defined(__powerpc64__)
+                ucontext->uc_mcontext.gp_regs[i]
 #elif defined(__sparc__) && defined(__arch64__)
                 ucontext->uc_mcontext.mc_gregs[i]
 #else
